kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/RawCustomResourceOperationsImpl.java

```diff
@@ -893,11 +893,18 @@ public Watch watch(Map<String, String> labels, ListOptions options, Watcher<Stri
       }
 
       try {
+        String nameFromObject = (String) metadata.get("name");
         // re-add for edit call
         if (originalResourceVersion != null) {
           metadata.put(RESOURCE_VERSION, originalResourceVersion);
+        } else {
+          // we get resourceVersion from existing object
+          // possible race condition if object has been deleted since the HTTP_CONFLICT exception?
```

Comment:
This implementation should match what we have in:

https://github.com/fabric8io/kubernetes-client/blob/b54d41f854d4a34ebf82cef8310f4226f702f834/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/HasMetadataOperation.java#L65-L74

In that case there is no check for deletion since the [CreateOrReplaceHelper#createOrReplace](https://github.com/fabric8io/kubernetes-client/blob/02826311061aa15633f7482397ae3ce11e26d34e/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/CreateOrReplaceHelper.java#L48) retries the operation 3 times.

A quick mitigation for some of the cases could be to check if the `metadata.deletionTimestamp` is not blank.

Reply:
I think this refactoring should belong to a different issue/pr because the existing code cannot replace the RCR without a `resourceVersion`.
The goal of this comment was just to highlight the potential issue for later.

Reply:
Don't worry, we can revisit this later.